src/Middleware/RequestQueue/src/SemaphoreWrapper.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.RequestQueue
+{
+    public class SemaphoreWrapper
```

Comment:
I missed the memo, why does this type exist?

Reply:
Because we wanted to start the process of submitting a PR early to get the experience. At some point, this will be renamed to RequestQueue.

Reply:
Ah, well there's a problem with having the namespace and a type name the same. So lets fix that now as it'll be much harder to change later and it'll make coding annoying.